### PR TITLE
fix(component): matchR cb value type infered if action is already handled in matchR or install

### PR DESCRIPTION
**Pull request by smehta91**
affects: @action-land/component

### CODE REVIEW

File: modules/component/src/componentNext.ts [hunk 1]
```diff
@@ -107,7 +119,7 @@ export class ComponentNext<P1 extends ComponentProps> {
       (a, s: any) => {
         const s2 = this._update(a, s) as any
         if (a.type === type) {
-          return cb(a.value as V, s2)
+          return cb(a.value as any, s2)
```
Comment (tusharmath) [inline modules/component/src/componentNext.ts]: we should a comment wherever we typecast.

File: modules/component/src/componentNext.ts [hunk 2]
```diff
@@ -92,12 +92,24 @@ export class ComponentNext<P1 extends ComponentProps> {
...
+        : LActionValueForType<iActions<P1>, T> extends Action<unknown>
+        ? LActionValueForType<iActions<P1>, T>
+        : V,
+      state: iState<P1>
+    ) => oState2
   ): iComponentNext<
     P1,
     {
       iActions: T extends LActionTypes<iActions<P1>>
-        ? Action<V & LActionValues<iActions<P1>>, T>
```
Comment (tusharmath) [inline modules/component/src/componentNext.ts]: add a type test.

File: typings/type.ts [hunk 3]
```diff
@@ -170,3 +177,29 @@ $(
 
 // $ExpectType { a: string; }
 $(ComponentNext.lift({a: ''}).render(_ => _.state)).oView
+
+// $ExpectType Action<Action<Action<unknown, "a">, "gc1"> | Action<number, "c1">, "child1"> | Action<never, "child2">
```
Comment (tusharmath) [inline typings/type.ts]: write simpler type-tests

File: modules/component/src/componentNext.ts [hunk 4]
```diff
@@ -39,6 +39,12 @@ type LActionTypes<A> = A extends Action<any, infer T> ? T : never
 type LActionValues<A> = A extends Action<infer V, any> ? V : never
 type LObjectValues<O> = O extends {[k: string]: infer S} ? S : unknown
 type LActionValueForType<A, T> = A extends Action<infer V, T> ? V : never
```
Comment (tusharmath) [inline modules/component/src/componentNext.ts]: we can make 

```
LActionValueForType<A, T, D = never> = A extends Action<infer V, T> ? V : D
```

File: modules/component/src/componentNext.ts (hunk 2 above)
Comment (smehta91) [reply modules/component/src/componentNext.ts]: added

File: modules/component/src/componentNext.ts (hunk 4 above)
Comment (smehta91) [reply modules/component/src/componentNext.ts]: this will not work when we pass `A` as  Action set.

**Pull request merged**